sdk/core/System.ClientModel/tests/Convenience/ClientRequestExceptionTests.cs

```diff
@@ -24,6 +25,20 @@ public void CanCreateFromResponse()
             exception.Message);
     }
 
+    [Test]
+    public async Task CanCreateFromAsyncFactory()
```

Comment:
I don't see any other tests added here, does this validate all the different scenarios about buffer and non buffering that the issue discusses?

Reply:
Yeah, good call-out.  We have a lot of test coverage in the Azure.Core integration branch (and full CI coverage that client functionality is correct).  Tracking getting to parity with ClientModel tests as part of https://github.com/Azure/azure-sdk-for-net/issues/40560

Reply:
Added a comment to that one to track these cases specifically.  Thanks @m-nash!